builtin/mainmenu/dlg_server_list_mods.lua

```diff
@@ -0,0 +1,96 @@
+-- Luanti
+-- Copyright (C) 2024 cx384
+-- SPDX-License-Identifier: LGPL-2.1-or-later
+
+local function get_formspec(dialogdata)
+	local TOUCH_GUI = core.settings:get_bool("touch_gui")
+	local server = dialogdata.server
+	local filter_prefix = dialogdata.filter_prefix
+
+	-- A wrongly behaving server may send ill formed mod names
+	local mods = {}
+	table.sort(server.mods)
+
+	if filter_prefix then
+		for _, mod in ipairs(server.mods) do
+			if mod:sub(0, #filter_prefix) == filter_prefix then
+				table.insert(mods, core.formspec_escape(mod))
+			end
+		end
+	else
+		local last_prefix
+		for i, mod in ipairs(server.mods) do
+			local prefix = mod:match("([^_]*_)")
+			if prefix and last_prefix == prefix then
+				mods[#mods] = "#BBBBBB▶ ".. core.formspec_escape(prefix) .. "*"
+			else
+				table.insert(mods, core.formspec_escape(mod))
+				last_prefix = prefix
+			end
+		end
+	end
+	dialogdata.mods = mods
+	mods = table.concat(mods, ",")
+
+	local prefix_button_label
+	if not filter_prefix then
+		prefix_button_label = "Expand all prefixes"
+	elseif filter_prefix == "" then
+		prefix_button_label = "Group by prefix"
+	else
+		prefix_button_label = "Show all mods"
+	end
+
+	local formspec = {
+		"formspec_version[8]",
+		"size[6,9.5]",
+		TOUCH_GUI and "padding[0.01,0.01]" or "",
+		"hypertext[0,0;6,1.5;;<global margin=5 halign=center valign=middle>",
+			fgettext("The $1 server uses " .. (server.gameid and "a game called $2 and " or "") ..
+					"the following mods:",
```

Comment:
Please only use string constants (without concatenation or ternary) in the first argument of `fgettext`. Otherwise `xgettext` does not correctly collect these strings for translation.